packages/nimble-components/src/table/index.ts

```diff
@@ -156,15 +171,65 @@ export class Table<
         this.virtualizer = new Virtualizer(this);
     }
 
-    public setData(newData: readonly TData[]): void {
-        this.setTableData(newData);
+    public async setData(newData: readonly TData[]): Promise<void> {
+        await this.processPendingUpdates();
```

Comment:
Can we capture somewhere maybe in a spec or maybe in a follow-up the theory of the table's state management that we have landed on?

i.e. properties are declarative and applied asynchronously in batches. methods that depend on properties are asynchronous, wait on pending batches and then resolve. also methods can apply state before execution